### PR TITLE
docs: Document requirements.txt for examples

### PR DESCRIPTION
**Pull request by mihaic**
Also add some missing requirements for some examples.

### CODE REVIEW

File: examples/funcalign/requirements.txt [hunk 1]
```diff
@@ -1,2 +1,3 @@
 matplotlib
+nilearn
```
Comment (yidawang) [inline examples/funcalign/requirements.txt]: Could you explain why this dependency just appeared, or why it was not here before?

File: examples/searchlight/requirements.txt [hunk 2]
```diff
@@ -0,0 +1,2 @@
+matplotlib
+nilearn
```
Comment (yidawang) [inline examples/searchlight/requirements.txt]: As above.

File: examples/funcalign/requirements.txt (hunk 1 above)
Comment (mihaic) [inline examples/funcalign/requirements.txt]: It should have been there already (i.e., this PR fixes a bug).

File: examples/searchlight/requirements.txt (hunk 2 above)
Comment (mihaic) [inline examples/searchlight/requirements.txt]: Same.

**Pull request merged**